torchdynamo/convert_frame.py

```diff
@@ -61,16 +62,32 @@ def fx_forward_from_src_skip_result(*args, **kwargs):
 
 def wrap_compiler_fn(compiler_fn):
     """Expand backend strings to functions"""
-    if compiler_fn == "inductor":
-        from torchinductor.compile_fx import compile_fx
+    if config.verify_correctness:
+        # wrap backend if verify_correctness is True
+        wrapper_compiler_fn = WrapperBackend(compiler_fn)
 
-        return compile_fx
-    elif isinstance(compiler_fn, str):
-        from .optimizations import BACKENDS
+        if compiler_fn == "inductor":
+            from torchinductor.compile_fx import compile_fx
+
+            wrapper_compiler_fn.backend = compile_fx
+        elif isinstance(compiler_fn, str):
+            from .optimizations import BACKENDS
+
+            wrapper_compiler_fn.backend = wrap_compiler_fn(BACKENDS[compiler_fn])
+
+        return wrapper_compiler_fn
 
-        return wrap_compiler_fn(BACKENDS[compiler_fn])
     else:
-        return compiler_fn
+        if compiler_fn == "inductor":
```

Comment:
Let's find a way not to duplicate this code here and above.  Perhaps a helper function?

Reply:
Use a _wrap_compiler_fn() as a helper function to make the code cleaner